src/main/java/seedu/address/model/RemindersImpl.java

```diff
@@ -111,6 +112,22 @@ public void deleteObsoleteReminders() {
         }
     }
 
+    /**
+     * When an event is being deleted, the associated reminder should be deleted as well.
+     * This is what the command aims to achieve.
+     */
+    public void deleteReminderOfEvent(Event eventToDelete) {
+        Reminder toDelete = null;
+        for (Reminder r: reminders) {
+            if (r.getEventToRemind().equals(eventToDelete)) {
+                toDelete = r;
+            }
+        }
+        if (toDelete != null) {
+            removeReminder(toDelete);
+        }
+    }
+
```

Comment:
This implementation would mean that editing a event and then deleting it would result in the same bug as the events are now not equal. Edited events are not updated on the reminder side. Rest looks ok for its intended purpose tho.

Reply:
This has been presumably resolved under changes to ModelManager.setEvent